### PR TITLE
Feature/id 34 donor id check

### PR DESCRIPTION
**Pull request by spinhalf-com**
# Purpose

_Briefly describe the purpose of the change._

Fixes ###-####

## Approach

_Explain how your code addresses the purpose of the change._

## Learning

_Any tips and tricks, blog posts or tools which helped you._

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [x] I have updated documentation where relevant
* [x] I have added tests to prove my work
* [ ] I have run an accessibility tool against the changes and applied fixes
* [ ] The team have tested these changes

### CODE REVIEW

File: service-front/module/Application/src/Contracts/OpgApiServiceInterface.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Contracts;
+
+interface OpgApiServiceInterface
```
Comment (gregtyler) [inline service-front/module/Application/src/Contracts/OpgApiServiceInterface.php:7]: FWIW we don't tend to use this pattern when interfaces only have one concrete. Which isn't to say "don't use it" but more "don't feel obliged to use it".

File: service-front/module/Application/view/layout/phase_banner.phtml
Comment (gregtyler) [inline service-front/module/Application/view/layout/phase_banner.phtml:1]: I don't think this partial is being used?

File: service-front/module/Application/view/application/pages/donor_id_check.phtml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            <li>
+                do they have the ID document with them?
+            </li>
+            <li>
+                Check the ID matches what is in Sirius (update as appropriate)
+            </li>
+        </ul>
+
+        <article class="moj-ticket-panel" aria-label="Sub navigation 1">
+            <section class="moj-ticket-panel__content moj-ticket-panel__content--blue" aria-label="Section 1">
```
Comment (gregtyler) [inline service-front/module/Application/view/application/pages/donor_id_check.phtml:32]: I'm not sure why the example in the design system uses article/section and aria-labels, but they don't semantically makes sense here so we should just use divs.
```suggestion
        <div class="moj-ticket-panel">
            <div class="moj-ticket-panel__content moj-ticket-panel__content--blue">
```

File: service-front/Dockerfile [hunk 3]
```diff
@@ -69,4 +69,4 @@ COPY phpcs.xml phpcs.xml
 COPY phpunit.xml phpunit.xml
 COPY psalm.xml psalm.xml
 
-FROM app AS production
+FROM app AS production
```
[the remark this answers is not in the document]
Comment (spinhalf-com) [reply service-front/Dockerfile:72]: Thanks - I enabled it in PHPStorm now

File: service-front/module/Application/view/application/pages/donor_id_check_with_table.phtml
Comment (gregtyler) [inline service-front/module/Application/view/application/pages/donor_id_check_with_table.phtml:1]: Sorry, just spotted: I don't think this file or `donor_lpa_check.phtml` are used?

File: service-front/module/Application/src/Factories/OpgApiServiceFactory.php [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+class OpgApiServiceFactory implements FactoryInterface
+{
+    /**
+     * @param ContainerInterface $container
+     * @param string                          $requestedName
+     * @param array<mixed>|null               $options
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null): OpgApiService
+    {
+        $guzzleClient = new Client(['base-url' => 'http://api-web']);
```
Comment (gregtyler) [inline service-front/module/Application/src/Factories/OpgApiServiceFactory.php:21]: Would still be good to move this to an env var but admittedly we can do that later

**Review by gregtyler — APPROVED** [review]

**Review by gregtyler — APPROVED** [review]

**Pull request merged**